src/fa.h

```diff
@@ -272,6 +277,27 @@ int fa_expand_nocase(const char *regexp, size_t regexp_len,
  */
 int fa_enumerate(struct fa *fa, int limit, char ***words);
 
+/* Print FA to OUT as a JSON file. State 0 is always the initial one */
+void fa_json(FILE *out, struct fa *fa);
+
+/* Returns 1 if the FA is deterministic and 0 otherwise */
+int fa_is_deterministic(struct fa *fa);
+
+/* Return the initial state */
+struct state *fa_state_initial(struct fa *fa);
+
+/* Return true if this is an accepting state */
+int fa_state_is_accepting(struct state *st);
+
+/* Return the next state; return NULL if there are no more states */
+struct state* fa_state_next(struct state *st);
+
+/* Return the number of transitions for a state */
+uint32_t fa_state_num_trans(struct state *st);
+
+/* Produce details about the i-th transition; return 0 on success, or -1 of there is no such transition */
+int fa_state_trans(struct state *st, uint32_t i, struct state **to, unsigned char *min, unsigned char *max);
```

Comment:
The `uint32_t` should be `size_t`. Mention in the comment that `to`, `min`, and `max` must not be `NULL`, and none of them will have been modified if `fa_state_trans` returns -1 (or maybe change the behavior to set `*to = NULL` in `fa_state_trans`).